docs/developer/extending/apps/extending-dashboard-with-apps.mdx

```diff
@@ -35,33 +46,185 @@ The example manifest below defines two extensions, one providing a custom produc
+
+- `GET` will open the URL similar to user opening the link in the new tab. Dashboard will append query params: `saleorApiUrl` and context fields (see "Context" below)
+- `POST` will open the URL using POST method. This is similar to submitting the HTML form by the user. The `body` will contain following fields:
+- `saleorApiUrl` - the URL of the saleor (starting with https, ending with `/graphql/`)
+- `appId` - ID of the app object
+- `accessToken` - short-living token that is intersection of app and logged user permissions
+- context fields — see below
+
+:::warning
+`accessToken` is short-living and can't be refreshed outside the Dashboard. You can likely communicate with Saleor during the request, but do not try to pass it to the frontend.
+
+To ensure seamless and safe user interaction with the UI, use App Page
+:::
+
+### `WIDGET` target (from 3.22)
+
+The `WIDGET` target stands for statically mounted iframes in various places around the dashboard. It will mount the frame similar to `APP_PAGE` and `POPUP`, but usually much smaller.
+
+Similar to `NEW_TAB` there are options allowing to choose the request method
+
+```json
+"options": {
+  "widgetTarget": {
+    "metod": "GET" // or "POST"
+  }
+}
+```
+If not provided, `GET` is the default choice for Dashboard.
+
+- `GET` will open the URL similar to user opening the link in the new tab. Dashboard will append query params: `saleorApiUrl` and context fields (see "Context" below). Additional fields will be passed to the iframe, similar to APP_AGE and POPUP, to help it render the UI (theme, feature flags, etc.)
+- `POST` will open the URL using POST method. This is similar to submitting the HTML form by the user. The `body` will contain the following fields:
+- `saleorApiUrl` - the URL of the saleor (starting with https, ending with `/graphql/`)
+- `appId` - ID of the app object
+- `accessToken` - short-living token that is intersection of app and logged user permissions. This token will be refreshed by the Dashboard in the background.
+- context fields — see below
+
+
+## Context
+
+Context fields are specific to the mount extension is being attached to. For example, on `CATEGORY_DETAILS_MORE_ACTIONS`, the context field will be `categoryId`.
+
+Context fields can be a `string`, usually an ID of the Saleor object, or a list of strings, usually when multiple Saleor items are selected by the user.
+
+For example
+- Order details page context will be `{orderId: "AABBCC"}`
+- Product list page context will be `{productIds: ["XXYYZZ", "BBCCDD"]}`
+
+Dashboard will provide contextual values depending on the mounting point. In the future more data can be passed to the context.
+
+### Context in GET operations
+
+`GET`-type operations, like rendering WIDGET using `options.widget.method: "GET"`, or `POPUP` will provide context fields as query params in URL. Other data like `theme` or feature flags will be also provided.
+
+You can easily access all the fields with JavaScript, like this:
```

Comment:
```suggestion
You can access all the fields with JavaScript, like this:
```
Maybe this is not tutorial but I would avoid words like `easily` https://css-tricks.com/words-avoid-educational-writing/#aa-10-easy